sentry_sdk/profiler.py

```diff
@@ -243,18 +263,24 @@ def __exit__(self, ty, value, tb):
         self.scheduler.stop_profiling()
         self._stop_ns = nanosecond_time()
 
-    def to_json(self, event_opt):
-        # type: (Any) -> Dict[str, Any]
+    def to_json(self, event_opt, options):
+        # type: (Any, Dict[str, Any]) -> Dict[str, Any]
         assert self._start_ns is not None
         assert self._stop_ns is not None
 
+        profile = self.scheduler.sample_buffer.slice_profile(
+            self._start_ns, self._stop_ns
+        )
+
+        handle_in_app_impl(
+            profile["frames"], options["in_app_exclude"], options["in_app_include"]
+        )
```

Comment:
If I understood the helper correctly, this does the following
- if no `in_app_exclude` or `in_app_include` are specified, it'll add `"in_app": True` to every frame
- if it is specified
  - frames that match `in_app_include` will have `"in_app": True` added to it
  - frames that match `in_app_exclude` will have `"in_app": False` added to it
  - other frames will not specify, but we should treat them as if they're not in_app frames